demo-client/src/main/java/com/illia/client/model/IMDbMovieParser.java

```
@@ -0,0 +1,80 @@
+package com.illia.client.model;
+
+import lombok.Data;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Component
+@Data
+public class IMDbMovieParser {
+
+    @Autowired
+    private IMDbMovieHolderImpl reportsHolder;
+
+    /**
+     * Title and date are mandatory, other attributes can be absent(null)
+     */
+    private static Pattern rowPattern = Pattern.compile("^([^;]+);(\\d+/\\d+/\\d+);(Color|Black and White)?;([a-zA-Z-]+)?;([a-zA-Z]+)?;([a-zA-Z ]+)?;([a-zA-Z0-9- ]+)?;([^;]+)?;([^;]+)?;(\\d+)?;(\\d+)?;(\\d+)?;(\\d+)?;(\\d+,?\\d*)?;(\\d+)?;(\\d+)?;(\\d+)?;(\\d+)?");
+
+
+    public List<IMDbMovieEntity> parseFile(File file) {
+
+        if (file.canRead()) {
+            try (var linesStream = new BufferedReader(new FileReader(file)).lines()) {
+                var reports = linesStream.skip(1)
+                        .map(this::parseRow)
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList());
+                reportsHolder.saveEntities(file.getName(), reports);
+                return reports;
+            } catch (Exception e) {
+                e.printStackTrace();
+            }
+        }
+        return List.of();
+    }
+
+    private IMDbMovieEntity parseRow(String row) {
+        Matcher matcher;
+        if ((matcher = rowPattern.matcher(row)).find()) {
+            return tryBuild(matcher);
+        } else{
+            return null;
```

if some rows cannot be parsed, we should throw an error